LanguageFeatures/Patterns/pattern_uses_A01_t01.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// It is a compile-time error if the expression in a guard clause in a switch
+/// case or if-case construct is not assignable to bool.
+///
+/// @description Check that it is a compile-time error if the expression in a
+/// guard clause is not assignable to bool.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+String test1(List<int> v) {
+  switch (v) {
+    case var [a, _] when a - 1:
```

Comment:
I think the spec parser is right: We can't derive a pattern of the form `var [a, _]` (it should be `[var a, _]`, assuming that we have surely covered the plain `var _` somewhere else).

Reply:
@eernstg please provide a link to that part of the spec which makes `var [a]` invalid in a matching context. I can't find it

Reply:
Oops, 2 weeks ago? Sorry about the delay. Anyway, here's an answer:

`var [a]` occurs after `case`, so it should be a `<guardedPattern>`. There's no `when`, so it should be a `<pattern>`. A `<pattern>` never starts with 'var' unless it is a `<variablePattern>`, and in that case it should have the form `'var' <identifier>`, which does not include `var [a]`.

So we can't do that.

Anyway, this change has actually been performed since then, so we probably agree on that already.